soc/arm/nordic_nrf/nrf53/soc.c

```diff
@@ -57,14 +58,18 @@ static int nordicsemi_nrf53_init(const struct device *arg)
 	key = irq_lock();
 
 #if defined(CONFIG_SOC_NRF5340_CPUAPP) && defined(CONFIG_NRF_ENABLE_CACHE)
-	/* Enable the instruction & data cache */
+	/* Enable the instruction & data cache.
+	 * This can only be done from secured code.
+	 * This is handled by the TF-M platform and cannot be disabled when TF-M
+	 * is enabled.
+	 */
```

Comment:
Did you compile for the wrong board perhaps? Enabling CACHE for nrf91 would probably lead to that, but otherwise I haven't seen any build warnings.

Reply:
Summary from offline discussion:
This was because NRF_ENABLE_CACHE had a dependency on TRUSTED_EXECUTION_NONSECURE.
This has now been changed so that NRF_ENABLE_CACHE is locked to enabled when BUILD_WITH_TFM is enabled.
This matches the configuration that cache is enabled by TF-M and no option in TF-M exists to disable it at the moment.